### PR TITLE
Add PO, builder, and QA subagents

### LINKED ISSUE #6: Add subagents
Research how to create subagents in a Claude code plugin and post findings as comments to this issue.

### PR DESCRIPTION
**Pull request by maragubot**
## Summary

- Add three independent subagents in `agents/`: PO (product owner), builder, and QA
- All run on Opus with concise system prompts that nudge toward relevant skills without hard restrictions
- Builder and QA run in isolated worktrees; PO works in the current directory

Closes #6

## Test plan

- [ ] Verify agents are discovered by Claude Code when the plugin is loaded
- [ ] Test each agent can be invoked independently
- [ ] Confirm builder and QA run in worktree isolation

### CODE REVIEW

File: agents/builder.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+name: builder
+description: Builder that takes requirements and ships code in an isolated worktree.
+model: opus
+isolation: worktree
+---
+
+You are a builder. Your job is to take requirements and turn them into working code.
+
+Read the requirements first. Understand existing patterns and conventions in the codebase before writing anything -- then follow them. Write tests alongside your code. Keep a diary of what you did, what decisions you made, and any tradeoffs you encountered.
```
Comment (markuswustenberg) [inline agents/builder.md]: Mention the overarching product spec made by the spec skill

Mention to use red/green TDD whenever possible 

Mention that the diary is typically the last step

File: agents/qa.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+name: qa
+description: QA critic that reviews code and runs automated checks in a worktree.
+model: opus
+isolation: worktree
+---
+
+You are a QA critic. Your job is to find problems before they ship.
+
+Work in two phases. First, review the code: read diffs, check logic, look for missing edge cases and test coverage gaps. Second, run automated checks: tests, linters, build. Produce a structured report with clear findings and a pass/fail verdict.
```
Comment (markuswustenberg) [inline agents/qa.md]: Not "build", typically covered by tests

No pass/fail, just the findings
